pkg/block/fetcher_test.go

```diff
@@ -969,9 +985,59 @@ func TestConsistencyDelayMetaFilter_Filter_0(t *testing.T) {
 		f := NewConsistencyDelayMetaFilter(nil, 30*time.Minute, reg)
 		testutil.Equals(t, map[string]float64{"consistency_delay_seconds": (30 * time.Minute).Seconds()}, extprom.CurrentGaugeValuesFor(t, reg, "consistency_delay_seconds"))
 
-		f.Filter(input, synced, false)
+		err := f.Filter(ctx, input, synced, false)
+		testutil.Ok(t, err)
 
 		testutil.Equals(t, float64(len(u.created)-len(expected)), promtest.ToFloat64(synced.WithLabelValues(tooFreshMeta)))
 		testutil.Equals(t, expected, input)
 	})
 }
+
+func TestIgnoreDeletionMarkFilter_Filter(t *testing.T) {
+	objtesting.ForeachStore(t, func(t *testing.T, bkt objstore.Bucket) {
+		ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+		defer cancel()
+
+		f := NewIgnoreDeletionMarkFilter(nil, bkt, 48*time.Hour)
+		now := time.Now()
+
+		shouldFetch := &metadata.DeletionMark{
+			ID:           ULID(1),
+			DeletionTime: now.Add(-15 * time.Hour).Unix(),
+			Version:      1,
+		}
+
+		shouldIgnore := &metadata.DeletionMark{
+			ID:           ULID(2),
+			DeletionTime: now.Add(-60 * time.Hour).Unix(),
```

Comment:
Ideally we would like to use mocked time, but I think given large delay durations (days) I think it's fine (:

Reply:
I am mocking time now. That is more cleaner. Resolving this thread